doc/sphinx/source/user_guide2/preprocessor.inc

```diff
@@ -40,6 +111,63 @@ The _area.py module contains the following preprocessor functions:
 * extract_named_regions: Extract a specific region from in the region cooordinate.
 
 
+1. extract_region
+-----------------
+
+This function masks data outside a rectagular region requested. The boundairies
+of the region are provided as latitude and longitude coordinates in the
+arguments:
+
+* start_longitude
+* end_longitude
+* start_latitude
+* end_latitude
+
+Note that this function can only be used to extract a rectangular region.
+
+
+2. zonal_means
+--------------
+
+The function calculates the zonal or meridional means. While this function is
+named `zonal_mean`, it can be used to apply several different operations in
+an zonal or meridional direction.
+ This function takes two arguments:
+
+* coordinate: Which direction to apply the operation: latitude or longitude
+* mean_type: Which operation to apply: mean, stdev, variance, median, min or max
+
+
+3. average_region
+-----------------
+
+This function calculates the average value over a region - weighted by the
+cell areas of the region.
+
+This function takes three arguments:
+coord1: the name of the coordinate in the first direction.
+coord2: the name of the coordinate in the second dimension.
+operator: the name of the operation to apply (default: mean).
+
+While this function is named `average_region`, it can be used to apply several
```

Comment:
Why not call it something like `region_statistic` instead if that is more accurate?

Reply:
We decided in #875 and #847 that we weren't going to rename the preprocessors yet - hence why I kept this name, despite it not fully describing the function. (Similarly for `zonal_mean`). I guess we'll change them all at some point, just not yet.

Reply:
Ok, maybe you can open an issue for it, so it is visible that this is planned?

Reply:
Okay, we should definately keep track of this. I think that Issue #847 is still open so I wrote a comment about this there. I think that #847 is still the best place for discussions about the preprocessor nomenclature.